network/wsNetwork_test.go

```diff
@@ -3757,3 +3759,106 @@ func TestWebsocketNetworkTelemetryTCP(t *testing.T) {
 	t.Log("closed detailsA", string(pcdA))
 	t.Log("closed detailsB", string(pcdB))
 }
+
+/*
+Exercises the updatePhonebookAddresses function, notably with different variations of valid relay and
+archival addresses.
+*/
+func TestUpdatePhonebookAddresses(t *testing.T) {
+	var netA *WebsocketNetwork
+
+	rapid.Check(t, func(t1 *rapid.T) {
+		netA = makeTestWebsocketNode(t)
+		relayPeers := netA.GetPeers(PeersPhonebookRelays)
+		assert.Equal(t, 0, len(relayPeers))
+
+		archivePeers := netA.GetPeers(PeersPhonebookArchivers)
+		assert.Equal(t, 0, len(archivePeers))
+
+		domainGen := rapidgen.Domain()
+
+		// Generate between 0 and N examples - if no dups, should end up in phonebook
+		domainsGen := rapid.SliceOfN(domainGen, 0, 200)
+
+		relayDomains := domainsGen.Draw(t1, "relayDomains").([]string)
+		archiveDomains := domainsGen.Draw(t1, "archiveDomains").([]string)
+		netA.updatePhonebookAddresses(relayDomains, archiveDomains)
+
+		// Check that entries are in fact in phonebook less any duplicates
+		dedupedRelayDomains := removeDuplicateStr(relayDomains)
+		dedupedArchiveDomains := removeDuplicateStr(archiveDomains)
+
+		relayPeers = netA.GetPeers(PeersPhonebookRelays)
+		assert.Equal(t, len(dedupedRelayDomains), len(relayPeers))
+
+		relayAddrs := make([]string, len(relayPeers))
+		for pi, peer := range relayPeers {
+			relayAddrs[pi] = peer.(HTTPPeer).GetAddress()
+		}
+		sort.Strings(relayAddrs)
+		sort.Strings(dedupedRelayDomains)
+		assert.Equal(t, dedupedRelayDomains, relayAddrs)
+
+		archivePeers = netA.GetPeers(PeersPhonebookArchivers)
+		assert.Equal(t, len(dedupedArchiveDomains), len(archivePeers))
+
+		archiveAddrs := make([]string, len(archivePeers))
+		for pi, peer := range archivePeers {
+			archiveAddrs[pi] = peer.(HTTPPeer).GetAddress()
+		}
+		sort.Strings(archiveAddrs)
+		sort.Strings(dedupedArchiveDomains)
+		assert.Equal(t, dedupedArchiveDomains, archiveAddrs)
+
+		// Generate fresh set of addresses with a duplicate from original batch if warranted,
+		// assert phonebook reflects fresh list / prior peers other than selected duplicate
+		// are not present
+		var priorRelayDomains = relayDomains
+		relayDomains = domainsGen.Draw(t1, "relayDomains").([]string)
+
+		// Randomly select a prior relay domain
+		if len(priorRelayDomains) > 0 {
+			priorIdx := rapid.IntRange(0, len(priorRelayDomains)-1).Draw(t1, "").(int)
+			relayDomains = append(relayDomains, priorRelayDomains[priorIdx])
```

Comment:
Why is `rapid` better than something more conventional?
```suggestion
			priorIdx := rand.Intn(len(priorRelayDomains))
			relayDomains = append(relayDomains, priorRelayDomains[priorIdx])
```

Reply:
I would keep rapid here, as when a test case fails it goes back re-running variations to minimize down to the exact inputs that seem to trigger the failure scenario. 

You are not wrong that random works, more for the traceability that the framework brings I would keep the existing option.